test/grasp/rcsfparser.jl

```diff
@@ -85,8 +85,16 @@ function parse_rcsf(filename)
             end
 
             config = Configuration(vcat(core_orbitals, orbitals), vcat(core_occupations, noccupations), sorted=true)
-            subshell_terms = map(x -> convert(Rational{Int}, x),
-                vcat(core_couplings, Vector{HalfInt}(orbcouplings)))
+            # I have no idea how to generate correct seniority numbers, even in the case where
+            # it does not matter (no degeneracy). So instead, I'll try to match the couplings
+            # to the intermediate terms generated by AtomicLevels.
+            subshell_terms = let its = intermediate_terms.(vcat(core_orbitals, orbitals), vcat(core_occupations, noccupations))
+                map(its, vcat(core_couplings, Vector{HalfInt}(orbcouplings))) do its, grasp_term
+                    idxs = findall(it -> it.term == grasp_term, its)
+                    @assert length(idxs) == 1 # make sure there is only one intermediate term with the coupling we're interested in
+                    its[first(idxs)]
+                end
+            end
```

Comment:
I fixed the GRASP tests, kinda. It would be handy to have a constructor `IntermediateTerm(coupling)`, which would automatically figure out the correct seniority number in the case when there's no ambiguity (or should it be `IntermediateTerm(orbital, coupling)`, because we'd need to know which orbital it is as well for the check?).

Reply:
Given a term, i.e. a `coupling`, and an orbital and its occupation (you need all three), you can find out _if_ a seniority number is valid, but you cannot say which one you should choose, i.e. there is no best default (unless there is only one valid option). This is precisely why they are generated _all together_ in `intermediate_terms`.

Reply:
In this case, there is only one seniority value (multiplicity is 1 for these terms). I just have no idea what the actual number is to actually construct the `IntermediateTerm` (it does not appear to be e.g. 0 or 1).

Reply:
Well, it is the occupation number of the subshell under consideration at which the term first appears (it is also, physically, [correspondent to the degree of unpaired electrons](https://en.wikipedia.org/wiki/Racah_seniority_number)). So, it is not simply a counter.

Reply:
I agree that it is a bit tricky to test the functionality of generating the correct seniority numbers, without actually reimplementing the logic in the tests. That is why I wondered if it was possible to get GRASP to print the numbers instead.